### PR TITLE
Enforce that only existing properties can be used

### PR DESCRIPTION
**Pull request by dstallenberg**

### CODE REVIEW

File: packages/core/lib/Configuration.ts [hunk 1]
```diff
@@ -20,15 +20,22 @@ import Yargs = require("yargs");
 import * as path from "path";
 import shell = require("shelljs");
 
-export let CONFIG: Readonly<ArgumentsObject>;
+export type RemoveIndex<T> = {
```
Comment (mitchellolsthoorn) [inline packages/core/lib/Configuration.ts:23]: Can you add a comment explaining the purpose of the type

File: packages/core/package-lock.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
       "license": "Apache-2.0",
       "dependencies": {
         "@dagrejs/graphlib": "2.1.4",
+        "@syntest/cfg-core": "^0.2.0-beta.0",
```
Comment (mitchellolsthoorn) [inline packages/core/package-lock.json:13]: Why did the lock file change?

**Review by mitchellolsthoorn — CHANGES_REQUESTED** [review]

File: packages/core/lib/Configuration.ts (hunk 1 above)
Comment (dstallenberg) [reply packages/core/lib/Configuration.ts:24]: Fixed.

File: packages/core/package-lock.json (hunk 2 above)
Comment (mitchellolsthoorn) [reply packages/core/package-lock.json:13]: We could always run it before Lerna version, so it also commits the lock file

Comment (mitchellolsthoorn) [reply packages/core/package-lock.json:13]: I created an issue for this #203

**Review by mitchellolsthoorn — APPROVED** [review]

**Pull request merged**